### DISCUSSION TITLE
notifications: Alert only if certain time passed?

### DISCUSSION THREAD

**Discussion by mubarak-j** [argoproj/argo-cd#9985] [Q&A]
I’m trying to avoid receiving false alerts/spam-like notifications when applications' health status change briefly e.g degraded/unknown <-> healthy often within a minute.

<img width="1166" alt="argocd-health-status-notification" src="https://user-images.githubusercontent.com/83465122/178900111-e0539e5a-dd08-471a-ab10-91b7971f5455.png">

I tried to add the following time condition to alert if the app has been in the current state for 5 mins or more but it’s not working, or maybe I’m using the wrong field `app.status.operationState.startedAt` here?

    trigger.on-health-degraded: |
      - description: Application has degraded
        oncePer: app.status.operationState.syncResult.revision
        send:
        - app-health-degraded
        when: app.status.health.status == 'Degraded' and time.Now().Sub(time.Parse(app.status.operationState.startedAt)).Minutes() >= 5
I'm curious if anyone solved this issue using argocd-notifications?

**Comment by dpotapovich** [comment]
Dear @alexmt,

can you provide some info, please?

**Reply by rimasgo** [reply]
I believe this should be there as well. 

@alexmt  the reason it does not work for you is because you're checking Health status and then taking the Data details from operationState (Sync operation for example) which is not relevant to Health status.

@dpotapovich  do you think it would be useful to have more details App Health information. 
For example if application is unhealthy for 10 minutes I would like to get notification. So all I would need to check is the health status and when that status changed. 

We have a APIService manifest which has label for kube-aggregrator to automanage itself. Which means that k8s API recycles the object. At that time ArgoCD losses track of the APIService and marks application as missing which sends a notification. Obviously the APIService stays missing just for a split second. Obviously I would not want to get the notification for this so I would need something what I described before. If application health status stays Missing for lets say 5 minutes send the notification. 

Please let me know if it makes sense.

**Reply by dpotapovich** [reply]
It is _**For example if application is unhealthy for 10 minutes I would like to get notification. So all I would need to check is the health status and when that status changed.**_ what I need too)

**Reply by dpotapovich** [reply]
I have the same situation with another service.

**Comment by kenske** [comment]
@mubarak-j were you able to get this to work?

**Reply by robmonct** [reply]
Interested too in this.

**Comment by FranAguiar** [comment]
Similiar use case here, I have an applicacion that sometimes takes a few minutes to be synced, but I want to know if it takes too much time to sync, a trigger to send a notification when is out of sync and "n" minutes would solve this. It is possible?

Thanks!

**Comment by booleanbetrayal** [comment]
We are seeing HCP Vault [inconsistency issues](https://developer.hashicorp.com/vault/docs/enterprise/consistency) throw [ExternalSecrets into a Degraded state](https://github.com/argoproj/argo-cd/blob/0a8a71e12c5010d5ada1fce37feb0d8add1c61d0/resource_customizations/external-secrets.io/ExternalSecret/health.lua), but are unsure of what the notification event payload looks like and what fields we could effectively implement this eval window against. We don't have the ability to configure the cluster prevent these sort of issues, and ESO doesn't really provide any failure conditioning, so we were hoping to filter at the ArgoCD level.

@alexmt - Is there an example of the event payload for Degraded events somewhere we can reference? Or maybe you know of a field off-hand?

**Reply by itaig-taranis** [reply]
facing the same issue

**Comment by yidongw** [comment]
interested in this as well

**Comment by mubarak-j** [comment]
related tickets: 
https://github.com/argoproj/notifications-engine/issues/99
https://github.com/argoproj/argo-cd/issues/12169
